pkg/build/builder/common.go

```diff
@@ -82,3 +92,134 @@ func updateBuildRevision(c client.BuildInterface, build *api.Build, sourceInfo *
 		glog.Warningf("An error occurred saving build revision: %v", err)
 	}
 }
+
+// randomBuildTag generates a random tag used for building images in such a way
+// that the built image can be referred to unambiguously even in the face of
+// concurrent builds of the same originalTag.
+func randomBuildTag(originalTag string) string {
```

Comment:
I'd like to have this be a shortened version of the build name rather than tied to the tag.  It needs a random suffix, but the inputs should be build name + build uid